app/src/store/stores/subServerStore.ts

```diff
@@ -0,0 +1,47 @@
+import { makeAutoObservable, runInAction } from 'mobx';
+import { Store } from 'store';
+import { SubServerStatus } from 'types/state';
+
+/** processed data for specific subservices we need to display in the UI */
+interface SubServers {
+  loop: SubServerStatus;
+  pool: SubServerStatus;
+}
+
+export default class SubServerStore {
+  private _store: Store;
+
+  loading = false;
+
+  subServers: SubServers = { loop: {} as SubServerStatus, pool: {} as SubServerStatus };
```

Comment:
I'd highly recommend initializing the fields in these obects with default values instead of empty object. You'll tend to run into hard to catch bugs when the types don't match the actual data stored in these variables.

```ts
  subServers: SubServers = {
    loop: {
      disabled: false,
      running: true,
      error: '',
    },
    pool: {
      disabled: false,
      running: true,
      error: '',
    },
  };
```

Reply:
Safer code is now being used :+1: 

https://github.com/lightninglabs/lightning-terminal/blob/106897df2a17595c8481c6e743d231124f469704/app/src/store/stores/subServerStore.ts#L16-L27